### PR TITLE
[IMP] sale_financial_risk: Bypass risk when payment transaction is authorized

### PR DESCRIPTION
**Pull request by carlosdauden**
@Tecnativa TT23520

### CODE REVIEW

**Review by pedrobaeza — COMMENTED** [review]
I miss the proper docstring on each method with the why you are passing that context. I wonder indeed why you need to overwrite 2 methods.

**Comment by carlosdauden** [comment]
We can overwrite those methods or this methods:
https://github.com/odoo/odoo/blob/a8689769afa22fb633f2e911f1a4dea6caa193d3/addons/sale/models/payment.py#L99
https://github.com/odoo/odoo/blob/a8689769afa22fb633f2e911f1a4dea6caa193d3/addons/sale/models/payment.py#L109

Or we can overwrite this methods:
https://github.com/odoo/odoo/blob/a8689769afa22fb633f2e911f1a4dea6caa193d3/addons/sale/models/payment.py#L82
https://github.com/odoo/odoo/blob/a8689769afa22fb633f2e911f1a4dea6caa193d3/addons/sale/models/payment.py#L138

As it is an innocuous context, I have decided to overwrite the most generic methods

**Comment by pedrobaeza** [comment]
But why 2 and not 1?

**Comment by carlosdauden** [comment]
Have you opened the links that I have put?

**Comment by pedrobaeza** [comment]
Yes, and I need a plain explanation, and it's what I'm requesting for docstrings. Having obscure overwrites or temporary links lead to not being sure when migrating this what to do.

**Comment by carlosdauden** [comment]
The confirmation of orders can come in two ways.
Invoice confirmation comes from one of those two ways.

bypass_risk in authorized transaction is pretty self explanatory, but you can make a suggestion for those docstrings and I will add them.

**Review by rafaelbn — APPROVED** [review]
Thanks :+1:

File: sale_financial_risk/models/payment.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2020 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    def _set_transaction_authorized(self):
```
Comment (chienandalu) [inline sale_financial_risk/models/payment.py:10]:
```suggestion
    def _set_transaction_authorized(self):
        """Bypass risk for sale confirmation triggered by this method"""
```

File: sale_financial_risk/models/payment.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        return super(PaymentTransaction, self.with_context(
+            bypass_risk=True))._set_transaction_authorized()
+
+    def _reconcile_after_transaction_done(self):
```
Comment (chienandalu) [inline sale_financial_risk/models/payment.py:15]:
```suggestion
    def _reconcile_after_transaction_done(self):
        """Bypass risk for sale confirmation and invoice creation triggered by this method"""
```

**Review by chienandalu — APPROVED** [review]

**Comment by carlosdauden** [comment]
Changes done

**Pull request merged**